src/bsp/src/stm32/src/interloc/include/interloc/Decawave.h

```diff
@@ -114,46 +114,40 @@ class Decawave {
      * @param buf Buffer to transmit
      * @param length Length of the data
      * @param rxAfterTxTimeUs Number of microseconds between the TX and the start of the RX
-     * @param frame [out] Variable in which all frame data will be put once packet is received
      * @param rxTimeoutUs Timeout in microseconds. 0 to block indefinitely
-     * @return True if successful, false otherwise.
+     * @return An UWBRxStatus if successful, false otherwise.
      */
-    bool transmitAndReceive(uint8_t* buf,
-                            uint16_t length,
-                            uint32_t rxAfterTxTimeUs,
-                            UWBRxFrame& frame,
-                            uint16_t rxTimeoutUs);
+    std::optional<UWBRxStatus> transmitAndReceive(uint8_t* buf,
```

Comment:
Why the optional here? Couldn't the error state be handled within the UWBRXStatus?